composer/callbacks/checkpoint_saver.py

```diff
@@ -358,60 +347,50 @@ def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
 
         is_deepspeed = is_model_deepspeed(state.model)
 
-        if is_deepspeed and '{rank}' not in self.filename.filename:
-            raise ValueError(f'Save filename {self.filename.filename} must have {{rank}} for deepspeed.')
-
-        # save the checkpoint to the filename
-        filename = self.filename.format(state, is_deepspeed)
-
         saved_path = checkpoint.save_checkpoint(
             state=state,
-            filename=filename,
+            filename=self.file_path,
             weights_only=self.weights_only,
         )
 
+        checkpoint_file_path = format_name_with_dist_and_time(self.checkpoint_file_path, state.run_name, state.timestamp)
+        checkpoint_file_path += ('.tar' if is_deepspeed and not is_tar(checkpoint_file_path) else '')
+
         if not saved_path:  # not all ranks save
             return
 
         if self.latest_filename is not None:
-            symlink = self.latest_filename.format(state, is_deepspeed)
-            os.makedirs(os.path.dirname(symlink), exist_ok=True)
+            symlink_name = format_name_with_dist_and_time(self.latest_filename, state.run_name, state.timestamp)
+            symlink_name += ('.tar' if is_deepspeed and not is_tar(self.latest_filename) else '')
```

Comment:
prefer a function for this deepspeed bit since it is called multiple times

Reply:
ok reverted back to using `PartialFilePath`